lib/dor-services.rb

```diff
@@ -31,29 +29,15 @@ def load_instance(pid)
     # @param [String] pid The object's PID
     # @return [Object] the ActiveFedora-modeled object
     def find(pid, opts = {})
-      opts[:rows] = 1 # we know we are going to just pay attention to the .first doc anyway
-      find_all(%(id:"#{pid}"), opts).first || load_instance(pid)
+      load_instance(pid)
     end
 
     # TODO: return enumerable and lazy load_instance
     # TODO: restrict fieldlist (fl) for non-:lightweight queries
     def find_all(query, opts = {})
       resp = SearchService.query query, opts
       resp.docs.collect do |solr_doc|
-        doc_version = solr_doc[INDEX_VERSION_FIELD].first rescue '0.0.0'
-        doc_version = Gem::Version.new(doc_version)
-        object_type = Array(solr_doc[ActiveFedora::SolrService.solr_name('objectType', :symbol)]).first
-        object_class = registered_classes[object_type] || ActiveFedora::Base
-        if opts[:lightweight] && doc_version >= Gem::Version.new('3.1.0')
```

Comment:
should we deprecate `find_all`? the only place I see it being used now is to find workflow objects by name.